### PR TITLE
uwsgi: fix compiling against php 8, bump to 2.0.20

### PR DESCRIPTION
**Pull request by gdamjan**
###### Motivation for this change

Current uwsgi 2.0.19.1 is not compatible with PHP 8.x, that support was added in 2.0.20
https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.20.html

Tested building with both php 7.4:
```
nix-build -E 'with import <nixpkgs> {}; uwsgi.override { php=pkgs.php74; plugins=[ "php" ]; }'
```
and php 8:
```
nix-build -E 'with import <nixpkgs> {}; uwsgi.override { plugins=[ "php" ]; }'
```

also patched this issue on top of 2.0.20:
https://github.com/unbit/uwsgi/issues/2356

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gdamjan** [comment]
ps.
I have a branch ready to patch nixos-21.11 too, although the commits should easily be cherry-picked.

File: pkgs/servers/uwsgi/default.nix [hunk 1]
```diff
@@ -91,12 +92,15 @@ stdenv.mkDerivation rec {
...
+  php8 = builtins.head (builtins.splitVersion php.version) == "8";
+  php8_no_version = ''sed -e "s/ + php_version//" -i plugins/php/uwsgiplugin.py'';
+
   postPatch = ''
     for f in uwsgiconfig.py plugins/*/uwsgiplugin.py; do
       substituteInPlace "$f" \
         --replace pkg-config "$PKG_CONFIG"
     done
-  '';
+  '' + (lib.optionalString php8 php8_no_version);
```
Comment (ajs124) [inline pkgs/servers/uwsgi/default.nix:103]: While it doesn't really change anything function-wise, this is kind of awkward to read, IMHO.

Why not just have this whole block in the `postPatch` as:
```
${lib.optionalString (lib.versionAtLeast php80.version "8") ''
  sed -e "s/ + php_version//" -i plugins/php/uwsgiplugin.py'';
''}
```
or 
```
+ (lib.optionalString (lib.versionAtLeast php80.version "8") ''
  sed -e "s/ + php_version//" -i plugins/php/uwsgiplugin.py'';
'')
```

Comment (gdamjan) [reply pkgs/servers/uwsgi/default.nix:103]: yeah, sure. I still lack the skills to write idiomatic nix. let me fix that

Comment (gdamjan) [reply pkgs/servers/uwsgi/default.nix]: changed it, does it look ok now?

**Review by ajs124 — APPROVED** [review]

**Comment by gdamjan** [comment]
let's merge this, since it needs to be backported to 21.11 too

**Pull request merged**